persistence/src/main/java/se/inera/intyg/intygstjanst/persistence/model/dao/impl/CertificateDaoImpl.java

```diff
@@ -190,6 +190,53 @@ public void removeCertificatesDeletedByCareGiver(Personnummer civicRegistrationN
         }
     }
 
+    @Override
+    public List<Certificate> findTestCertificates(LocalDateTime from, LocalDateTime to) {
+        final var criteriaBuilder = entityManager.getCriteriaBuilder();
+        final var query = criteriaBuilder.createQuery(Certificate.class);
+        final var root = query.from(Certificate.class);
+
+        final List<Predicate> predicates = new ArrayList<>();
+
+        predicates.add(
+            criteriaBuilder.isTrue(
+                root.get("testCertificate")
+            )
+        );
+
+        if (from != null) {
+            predicates.add(
+                criteriaBuilder.greaterThanOrEqualTo(
+                    root.get("signedDate"), from
+                )
+            );
+        }
+
+        if (to != null) {
+            predicates.add(
+                criteriaBuilder.lessThan(
+                    root.get("signedDate"), to
+                )
+            );
+        }
+
+        query.where(predicates.toArray(new Predicate[predicates.size()]));
```

Comment:
Är det tänkt att alla ska hämtas om from och to är null?

Reply:
Såg nu att det fanns kontroller på att to och from inte var null i api-anropet.

Reply:
Jag tänkte att DAO-implementationen kan ge möjlighet att användas med eller utan datum. För att öka återanvändningen av den. Men för denna jira så används inte det.